pandas/_libs/window.pyx

```diff
@@ -654,16 +654,21 @@ cdef inline void add_var(double val, double *nobs, double *mean_x,
                          double *ssqdm_x) nogil:
     """ add a value from the var calc """
     cdef double delta
-
```

Comment:
doing
```
if val != val:
    return

# rest of code
```

might work (and be more clear here)

Reply:
Great suggestion, sadly didn't work

Reply:
Sometimes I see compiler warnings about tautologically false comparisons.  Is there a chance the compiler threw out this check?